<launch>
<!-- hehe -->
    <!-- args for all nodes -->
    <!--<arg name="log_path" default="~/ros_farmer_logs"/>-->
    <arg name="arduino_baudrate" default="57600"/>
    <arg name="arduino_port" default="/dev/serial/by-id/usb-1a86_USB2.0-Serial-if00-port0"/>

    <arg name="led_baudrate" default="19200"/>
    <arg name="led_device_port" default="/dev/serial/by-id/usb-Silicon_Labs_CP2102_USB_to_UART_Bridge_Controller_0001-if00-port0"/>

    <arg name="relay_number_of_channels" default="8"/>
    <arg name="relay_mapping_string" default="N2_VALVE,AIR_PUMP_1,AIR_PUMP_2,COOLER_1,AIR_VALVE_1,AIR_VALVE_2,NDIR_PUMP,EMPTY"/>
    <arg name="raw_relay_topic_name" default="relay_1_sub"/>

    <arg name="control_relay_service_name" default="relay_device"/>
    <arg name="control_led_service_name" default="led_device"/>
    <arg name="control_sba5_service_name" default="sba5_device"/>
    <arg name="control_start_mode" default="life_support"/>

    <!-- nodes -->
    <node pkg="ros_farmer_pc" type="control_system.py" name="control_node">
        <param name="control_relay_service_name" value="$(arg control_relay_service_name)"/>
        <param name="control_led_service_name" value="$(arg control_led_service_name)"/>
        <param name="control_sba5_service_name" value="$(arg control_sba5_service_name)"/>
        <!-- set LSS mode -->
        <param name="control_start_mode" value="$(arg control_start_mode)"/>
    </node>

    <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
        <param name="port" value="$(arg arduino_port)"/>
        <param name="baud" value="$(arg arduino_baudrate)"/>
    </node>

  <node pkg="ros_farmer_pc" type="led_device.py" name="led_device">
    <param name="led_port" value="$(arg led_device_port)"/>
    <param name="led_baudrate" value="$(arg led_baudrate)"/>
  </node>

  <node pkg="ros_farmer_pc" type="relay_device.py" name="relay_device">
    <param name="relay_number_of_channels" value="$(arg relay_number_of_channels)"/>
    <param name="relay_mapping_string" value="$(arg relay_mapping_string)"/>
    <param name="raw_relay_topic_name" value="$(arg raw_relay_topic_name)"/>
  </node>
<!--
   <node pkg="ros_farmer_pc" type="sba5_device.py" name="sba5_device">
    <param name="_sba5_port" value="$(arg sba5_device_port)"/>
    <param name="_sba5_baudrate" value="$(arg sba5_baudrate)"/>
    <param name="_sba5_timeout" value="$(arg sba5_timeout)"/>
    <param name="_sba5_calibration_time" value="$(arg sba5_calibration_time)"/>
  </node>
  -->
    <node pkg="ros_farmer_pc" type="watchdog.py" name="watchdog" required="true"/>
</launch>